crates/sui-core/src/unit_tests/move_integration_tests.rs

```diff
@@ -493,9 +493,8 @@ async fn test_object_owning_another_object() {
         vec![]
     };
     assert!(effects.status().is_ok());
-    assert_eq!(events.len(), 7);
```

Comment:
How did the extra event go away? Change in object size?

Reply:
This is a mystery to me. I took a look to make sure that the TODO below is correct, and confirmed that before this PR, there is an extra (and incorrect) `CoinBalanceChange` event with a value of -1. I'm not sure why this happened (probably a bug in the code that creates these events?), or why this PR fixes it. Don't feel great about it, but my understanding is that @patrickkuo is eliminating these events from the effects, so am inclined not to dig more here.

Reply:
yes I am nuking it as we speak

Reply:
That spurious event cropped up when we started using address32 FWIW.